### PR TITLE
Added a test runner for bulk update

### PR DESCRIPTION
**Pull request by ke4**
Changes:
- Add integration test for the spreadsheet update process.

The test is going through this process:
- Import the SS2 spreadsheet to ingest
- Modify some values of the imported projects with REST calls
- Download the spreadsheet of the project
- Modify some values in the downloaded spreadsheet
- Upload the modified spreadsheet
- Assert all the modified values (changes with REST calls and in the downloaded spreadsheet)

### CODE REVIEW

**Review by amnonkhen — APPROVED** [review]
This looks OK.
The question, IMO, is how easy it is to maintain the test, add an additional scenario, and investigate test failures.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -46,13 +46,25 @@ test_big_submission:
...
     - ingest-integration-tests
 
+test_bulk_update:
+  stage: test
+  script:
+     - python -m unittest tests.test_ingest.TestRun.test_bulk_update
+  only:
+    - dev
+    - staging
+    - add_test_for_bulk_update
```
Comment (aaclan-ebi) [inline .gitlab-ci.yml]: please don't forget to change later! i also think it's okay to include this test in production, it's not exporting anything so it should be fine

File: tests/runners/bulk_update_runner.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        self.__upload_modified_spreadsheet(updated_spreadsheet_path)
+        self.__validate_modifications(updated_project_description, updated_contributor_name, updated_bsd_accession,
+                                      updated_project_title, updated_biomaterial_name)
+
+    def __import_spreadsheet_with_files(self, dataset_fixture, project_uuid=None):
+        self.dataset = dataset_fixture
+        self.__upload_spreadsheet_and_create_submission(dataset_fixture, project_uuid)
+        self.submission_manager = SubmissionManager(self.submission_envelope)
+        self.submission_manager.get_upload_area_credentials()
+        self.submission_manager.stage_data_files(self.dataset.config['data_files_upload_area_uuid'])
```
Comment (aaclan-ebi) [inline tests/runners/bulk_update_runner.py:49]: I believe you do not need to upload files to test metadata updates

File: tests/runners/bulk_update_runner.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        self.project_content = project_payload.get('content')
+        return project_payload
+
+    def __get_biomaterial_content_by_id(self):
+        biomaterials_payload = \
+            self.bulk_update_manager.get_entities_by_submission_id_and_type(self.submission_id, 'biomaterials')
+        self.biomaterial_content_by_id = {
+            self.bulk_update_manager.get_id_from_entity(payload): payload.get('content')
+            for payload in biomaterials_payload
+        }
```
Comment (aaclan-ebi) [inline tests/runners/bulk_update_runner.py:83]: interesting syntax, i didn't know you could do this

File: tests/runners/bulk_update_runner.py (hunk 2 above)
Comment (aaclan-ebi) [reply tests/runners/bulk_update_runner.py:49]: removing it will also save time for running this test

**Review by jacobwindsor — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (ke4) [reply .gitlab-ci.yml]: ah, yep, tx. I remembered it last night, but it was too late and wanted to do today, then I forgot. Many thanks. I am going to changed it back now.

File: tests/runners/bulk_update_runner.py (hunk 2 above)
Comment (ke4) [reply tests/runners/bulk_update_runner.py:49]: Are you sure? It only take 1-2 seconds

File: tests/runners/bulk_update_runner.py (hunk 3 above)
Comment (ke4) [reply tests/runners/bulk_update_runner.py:83]: Well, sometimes you can even learn from me, too 😜

File: .gitlab-ci.yml (hunk 1 above)
Comment (ke4) [reply .gitlab-ci.yml]: Done ✅

File: tests/runners/bulk_update_runner.py (hunk 2 above)
Comment (aaclan-ebi) [reply tests/runners/bulk_update_runner.py:49]: yeah the test files are very small, i guess no big impact/not much processing is saved if you remove this

**Review by aaclan-ebi — APPROVED** [review]

**Pull request merged**